media_driver/linux/common/os/mos_gpucontext_specific.cpp

```diff
@@ -275,10 +318,14 @@ void GpuContextSpecific::Clear()
     MOS_SafeFreeMemory(m_attachedResources);
     MOS_SafeFreeMemory(m_writeModeList);
     MOS_SafeFreeMemory(m_createOptionEnhanced);
-    if (m_i915Context)
+
+    for (int i=0; i<MAX_ENGINE_INSTANCE_NUM; i++)
     {
```

Comment:
Hmm.. is number of created context really relating to max number of engine instances or it is actually one to three? (single pipe, master, slave)

Reply:
For ICL, 3 is enough. But same as mos_set_context_param_bond_slave, it needs to be refined to cover more slaves. It should be one single pipe, one master, and MAX_ENGINE_INSTANCE_NUM-1 slaves: for (int i=0; i<=MAX_ENGINE_INSTANCE_NUM; i++) {...}